src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -109,170 +142,122 @@ public static Regex GetOrAdd(string pattern, RegexOptions options, TimeSpan matc
 
         private static bool TryGet(Key key, [NotNullWhen(true)] out Regex? regex)
         {
-            Node? cachedRegex = s_cacheFirst;
-            if (cachedRegex != null)
+            long lastAccessedStamp = 0;
+
+            // We optimize for repeated usage of the same regular expression over and over,
+            // by having a fast-path that stores the most recently used instance.  Check
+            // to see if that instance is the one we want; if it is, we're done.
+            Node? lastAccessed = s_lastAccessed;
+            if (lastAccessed != null)
             {
-                if (cachedRegex.Key.Equals(key))
+                if (lastAccessed.Key.Equals(key))
                 {
-                    regex = cachedRegex.Regex;
+                    regex = lastAccessed.Regex;
```

Comment:
Why don't we update the `lastAccessed.LastAccessStamp` in this case?

Reply:
Exactly.

We also don't update it in Add in the case where it's found to already be in the dictionary.  In that case, we don't bother to spend the time doing it because we just checked in TryGet, which means it was only just added to the dictionary and its time was recently set.  My philosophy here is if there's no concurrency, then this doesn't matter, and if there is concurrency, all of this ordering is approximate anyway, so we don't need to be "perfect", since we effectively can't be.  The end goal should just be to get rid of a less used instance when we need to drop one; it doesn't need to be the absolute oldest instance.  If we're dropping from the cache frequently, the app has bigger problems.